### PR TITLE
New stackdriver_agent plugin, which sends metadata about the stackdriver agent itself

### PR DESCRIPTION
**Pull request by coreykosak**
~~**\* **DO NOT SUBMIT** **\* yet.... This is to give you folks a look at it first~~

### CODE REVIEW

**Comment by coreykosak** [comment]
Jeremy, because Igor is out of town, could you kindly take a look at this with an eye towards getting it submitted soon? Thanks.

File: src/stackdriver_agent.c [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        sagt_submit_derive("uptime", NULL, now, interval, uptime, NULL);
+    }
+
+    // memory used
+    {
+        FILE *f = fopen("/proc/self/statm", "r");
+        if (f)
+        {
+            size_t vm = 0;
+            if (fscanf(f, "%zu", &vm))
```
Comment (katzj) [inline src/stackdriver_agent.c]: I think we do but just to confirm, we want VmSize and not VmRss.

File: src/stackdriver_agent.c [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        {
+            sagt_submit_derive("api_request_count", "connectivity_failures", now, interval, value, NULL);
+        }
+        if (uc_meta_data_get_unsigned_int(&vl, SAGT_API_REQUESTS_ERRORS, &value) == 0)
+        {
+          sagt_submit_derive("api_request_count", "errors", now, interval, value, NULL);
+        }
+    }
+
+    // configured_plugins: TODO
```
Comment (katzj) [inline src/stackdriver_agent.c]: Maybe make this comment a little more verbose?

**Comment by katzj** [comment]
Two minor things but LGTM.

File: src/stackdriver_agent.c (hunk 2 above)
Comment (coreykosak) [inline src/stackdriver_agent.c]: Eh, I think I'll delete it. It was meant to be a placeholder for some future work where we send information about configured plugins, but I don't think there's any need to leave a placeholder there.

**Comment by coreykosak** [comment]
PTAL

**Comment by katzj** [comment]
lgtm

**Pull request merged**